plugin/interface_generator/main.go

```diff
@@ -353,7 +380,7 @@ func (api *apiTimerLayer) recordTime(startTime timePkg.Time, name string, succes
 func (api *apiTimerLayer) {{.Name}}{{funcStyle .Params}} {{funcStyle .Return}} {
 	startTime := timePkg.Now()
 	{{ if .Return }} {{destruct "_returns" .Return}} := {{ end }} api.apiImpl.{{.Name}}({{valuesOnly .Params}})
-	api.recordTime(startTime, "{{.Name}}", true)
+	api.recordTime(startTime, "{{.Name}}", {{ if .Return }} {{ shouldRecordSuccess "_returns" .Return }} {{ else }} true {{ end }})
```

Comment:
I was wondering if we could refactor this whole if-else to another helper function ? Seems like it's completely duplicated to me.